api/server/sdk/volume_ops.go

```diff
@@ -491,3 +506,49 @@ func setSpecBool(current, req bool, reqSet interface{}) bool {
 	}
 	return current
 }
+
+func GetEnforcedVolSpecs(locator *api.VolumeLocator, source *api.Source, spec *api.VolumeSpec) (*api.VolumeSpec, error) {
+	if locator != nil {
+		if val, ok := locator.VolumeLabels["enforce_disable"]; ok {
+			if val == "true" {
+				return spec, nil
+			}
+		}
+	}
+	storPolicy, err := policy.Inst()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Unable to get storage policy instance %v", err)
+	}
+
+	// check if custom policy passed with volume
+	if spec.GetStoragePolicy() != "" {
+		inspReq := &api.SdkOpenStoragePolicyInspectRequest{
+			Name: spec.GetStoragePolicy(),
+		}
+		customPolicy, err := storPolicy.Inspect(context.Background(), inspReq)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Failed to get %s policy details :%v", inspReq.Name, err)
+		}
+		// update spec with custom storage policy
+		updatedSpec := mergeVolumeSpecs(spec, customPolicy.GetStoragePolicy().GetPolicy())
+		return updatedSpec, nil
+	}
+
+	// apply default policy to volume
```

Comment:
This is interesting. Please send a note to @jvinod and ask, if a user enforces a policy, I'm thinking that it will apply to ALL requests, no matter what they choose, right?